client/my-sites/checkout/checkout/checkout.jsx

```diff
@@ -365,6 +366,10 @@ export class Checkout extends React.Component {
 			return `/checklist/${ selectedSiteSlug }`;
 		}
 
+		if ( this.props.isJetpack && isEnabled( 'jetpack/checklist' ) ) {
+			return `/plans/my-plan/${ selectedSiteSlug }?thank-you`;
+		}
```

Comment:
This will not set anything up automatically in any env that has `jetpack/checklist` enabled. Maybe we should omit before landing (#26637 and #26690).